ssl/record/rec_layer_s3.c

```diff
@@ -356,7 +356,7 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, size_t len,
      * promptly send beyond the end of the users buffer ... so we trap and
      * report the error in a way the user will notice
      */
-    if (len < s->rlayer.wnum) {
+    if ((len < s->rlayer.wnum) || ((tot != 0) && (len < (tot + s->rlayer.wpend_tot)))) {
```

Comment:
1) (tot != 0) check is added because wpend_tot is not resetted to 0 in ssl3_write_pending after wb->left becomes 0. So either we should add (tot != 0) or (wb->left != 0) here.